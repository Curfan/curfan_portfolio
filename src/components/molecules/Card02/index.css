.card02 {
	display: flex;
	justify-content: center;

	& > .wrapper {
		display: flex;
		justify-content: space-between;
		width: 100%;
		max-width: 950px;

		& > .image-wrapper {
			width: 100%;
			max-width: 330px;
			margin-right: 24px;

			& > .image {
				display: flex;
				flex-direction: column;
				justify-content: center;
				align-items: center;
				width: 100%;
				padding-top: 100%;
				background-size: cover;
				background-position: center;
			}
		}

		& > .content {
			min-width: 400px;

			& > .name {
				font-size: 36px;
				letter-spacing: 0.05em;
				margin-bottom: 54px;
			}

			& > .title {
				font-size: 18px;
				line-height: 26px;
				letter-spacing: 0.05em;
				margin-bottom: 8px;
			}

			& > .time {
				font-size: 18px;
				line-height: 26px;
				letter-spacing: 0.05em;
			}

			& > .notes {
				margin-top: 80px;

				& > .note {
					font-size: 18px;
					line-height: 26px;
					letter-spacing: 0.05em;
					margin-bottom: 16px;
				}
			}
		}
	}
}
