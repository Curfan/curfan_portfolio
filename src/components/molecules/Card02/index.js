import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

const Card02 = ({ className, photo, name, title, time, notes = [] }) => (
	<div className={classnames(styles.card02, className)}>
		<div className={styles.wrapper}>
			<div className={styles.imageWrapper}>
				<div
					className={styles.image}
					style={{
						backgroundImage: `url(${photo})`,
					}}
				/>
			</div>
			<div className={styles.content}>
				<div className={styles.name}>{name}</div>
				<div className={styles.title}>{title}</div>
				<div className={styles.description}>{time}</div>
				<div className={styles.notes}>
					{notes.map(note => (
						<div key={note} className={styles.note}>
							{note}
						</div>
					))}
				</div>
			</div>
		</div>
	</div>
);

export default Card02;
