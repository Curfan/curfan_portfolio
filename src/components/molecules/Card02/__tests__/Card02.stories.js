import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, object } from '@storybook/addon-knobs/react';

import Card02 from 'components/molecules/Card02';

const stories = storiesOf('molecules/Card02', module);

stories.add('__interactive', () => (
	<Card02
		photo={text('photo', 'https://fakeimg.pl/300/?text=PHOTO')}
		name={text('name', 'Name')}
		title={text('title', 'title')}
		time={text('time', 'time')}
		notes={object('notes', ['notes...', 'notes...', 'notes...'])}
	/>
));
