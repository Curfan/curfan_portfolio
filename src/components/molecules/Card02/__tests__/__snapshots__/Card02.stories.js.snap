// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="card02"
  >
    <div
      class="wrapper"
    >
      <div
        class="imageWrapper"
      >
        <div
          class="image"
          style="background-image: url(https://fakeimg.pl/300/?text=PHOTO);"
        />
      </div>
      <div
        class="content"
      >
        <div
          class="name"
        >
          Name
        </div>
        <div
          class="title"
        >
          title
        </div>
        <div
          class="description"
        >
          time
        </div>
        <div
          class="notes"
        >
          <div
            class="note"
          >
            notes...
          </div>
          <div
            class="note"
          >
            notes...
          </div>
          <div
            class="note"
          >
            notes...
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
